WordPress/wporg-mu-plugins
# Reduce font sizes

Fixes https://github.com/WordPress/wporg-main-2022/issues/95


The font subsets here are based on [Google Fonts](https://fonts.googleapis.com/css2?family=Inter&display=swap) and the [post](https://cloudfour.com/thinks/font-subsetting-strategies-content-based-vs-alphabetical/#footnote-2). And if we download one of the font files in [Google Fonts](https://fonts.googleapis.com/css2?family=Inter&display=swap), for example, [Latin](https://fonts.gstatic.com/s/inter/v12/UcCO3FwrK3iLTeHuS_fvQtMwCp50KnMw2boKoduKmMEVuLyfAZ9hiJ-Ek-_EeA.woff2), we can find that it's much smaller than ours (17kB vs. 78kB), that's because it only provides one font-weight while ours provide all weights that `Inter` has supported. 

We could also subset the fonts to a particular font weight as well for the home page (which has higher traffic and it's the entrance of the whole w.org) to pursue better performance.

A font-subset script was also committed in this PR, not sure if it's suitable to put it here, let me know if there's a better place for it.



### Screenshots
#### Sizes are around one-third of the original ones.

| Before | After |
| ------- | ----- |
| <img width="823" alt="image" src="https://user-images.githubusercontent.com/18050944/187217248-96c3a1ed-0bb5-4a7a-b029-a98ed2e51399.png"> | <img width="822" alt="image" src="https://user-images.githubusercontent.com/18050944/187210510-f4e41a25-82cd-47b6-a829-a91af3bce667.png"> |

#### Italic, different font weights, and Greek/Cyrillic/Latin are still provided.
https://user-images.githubusercontent.com/18050944/187210593-6185e308-4d38-43b1-b1f5-5d21e3777f26.mp4

### How to test the changes in this Pull Request:

1. Build `wporg-mu-plugins` in `wporg-main-2022`.
2. Go to homepage and check if the fonts remain the same.
3. Press Edit Site, add/edit a Heading block, and open devTools to check if the sizes are really smaller than the old ones.
4. Change the font family to Inter/EB Garamond in the editor and see if it still provides _italic_, **different font weights**, and Greek, Latin, and Cyrillic text.
